cmd/block-generator/generator/generate.go

```diff
@@ -637,3 +650,82 @@ func (g *generator) generateAssetTxn(round uint64, intra uint64) (transactions.S
 
 	return signTxn(txn), transactions.ApplyData{}, nil
 }
+
+func (g *generator) WriteAccount(output io.Writer, accountString string) error {
+	addr, err := basics.UnmarshalChecksumAddress(accountString)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal address: %w", err)
+	}
+
+	idx := accountToIndex(addr)
+
+	// Asset Holdings
+	assets := make([]generated.AssetHolding, 0)
+	createdAssets := make([]generated.Asset, 0)
+	for _, a := range g.assets {
```

Comment:
The change to make asset holdings a pointer, and adding a reference instead of a bool was done to make this loop faster. Previously everything was optimized for creating / deleting random assets/holdings with no concern to lookups.